envoy/admin/v2/admin.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+package envoy.admin;
+
+import "google/protobuf/any.proto";
+import "envoy/api/v2/rds.proto";
+
+import "gogoproto/gogo.proto";
+
+// [#protodoc-title: ConfigDump]
+
+// The /config_dump admin endpoint uses this wrapper message to maintain and serve arbitrary
+// configuration information from any component in Envoy.
+message ConfigDump {
+  // This map is serialized and dumped in its entirety at the /config_dump endpoint.
+  //
+  // Keys should be a short descriptor of the config object they map to. For example, envoy's HTTP
+  // routing subsystem might use "routes" as the key for its config, for which it uses the message
+  // RouteConfigDump as defined below. In the future, the key will also be used to filter the output
+  // of the /config_dump endpoint.
+  map<string, google.protobuf.Any> configs = 1 [(gogoproto.nullable) = false];
```

Comment:
@louiscryan advocated that we maintain these in a format suitable for replay. For that to work, I think the string here should be the type URL. We need to think how this is going to work with https://github.com/envoyproxy/data-plane-api/pull/527. We should have a doc for this by later this week. I think we might want this to be something like `type URL -> resource name -> (resource proto x version)` or something like this. It depends what we settle on for versioning in incremental xDS.

Reply:
Yeah +1 it would be nice to just have version be part of this, then we can also solve https://github.com/envoyproxy/envoy/issues/2172.

Reply:
I originally had that but saw that Any has the `type_url`, as in https://gist.github.com/jsedgwick/3a46408a9728ccef9a63d32b4c463c2b
Now, it's missing the version - is that a deficiency of Any?
So currently it's more like `resource name -> type URL -> resource proto'. Does it make more sense to group `type_url` and the actual resource, just like with Any? The keep the resource name top level and bundle in the version at the end.

Anyways, bikeshedding, we can just hold on that until you guys settle on versioning.

Reply:
You don't need to wait on the versioning. Just update the map target to be a struct which contains a string version and an Any object.

Reply:
Hold on, what exactly is the version we're encoding? Or is that what's pending resolution? Shall I just fill the field with empty strings?

Reply:
No, you are filling it with https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/discovery.proto#L60

Reply:
Which will resolve envoyproxy/envoy#2172

Reply:
I see. But in this case, those belong to `RouteConfiguration`, not `RouteConfigDump`, no? Which version would be pair with the `RouteConfigDump` in the `Any`?

Reply:
RDS is a bit of a tricky one, since technically without ADS there can be multiple RDS streams, 1 per HTTP connection manager. In that case, yes, the version likely needs to go with the route configuration. For the other cases it could go at the top level (CDS, LDS, etc.).

Another way of doing it, probably better, would be to track the discovery response version that carried a discrete cluster, route, listener, etc. and print that along with each object. This would work for incremental updates better. So maybe that's that we should do. In light of that I would take a look at the message structure and figure out a nesting that works. I don't think you need to wait until the incremental stuff is decided.

Reply:
🆗

Reply:
@mattklein123 One of the possible incremental xDS option is to have a version at resource level. Let's discuss there and circle back here.

Reply:
@htuch OK SGTM.